xbmc/settings/DisplaySettings.cpp

```diff
@@ -792,6 +792,26 @@ void CDisplaySettings::SettingOptionsMonitorsFiller(SettingConstPtr setting, std
     }
     list.push_back(std::make_pair(monitors[i], monitors[i]));
   }
+#elif defined(HAVE_WAYLAND)
```

Comment:
I didn't want to accidentally break something, so I made it Wayland-specific. But I agree that it might be good to do this on X11, too, if it also has this problem. How does X11 react at the moment when you set a nonexistant monitor in the config and go to the settings screen?